app/controllers/carts_controller.rb

```diff
@@ -25,9 +25,29 @@ def destroy
     end
   end
 
+  def checkout
+    create_new_order unless unfinished_checkout?
```

Comment:
czy to nie jest goto? :) Nie podoba mi sie rowniez, ze redirect jest wywolywany z jakiejs prywatnej metody.

Reply:
Trochę to desperacko zrobiłem bo nie wiedziałem jak uciec od piekła ifów. Postanowiłem całkiem to zmienić w nowym commicie i dać całą pracę wykonać serwisowi.